4－7統計検定で使用したデータ（student_data_merge）をつかって、以下の問いに答えてください。

１，それぞれの欠席数（absences）について、差があるといえるでしょうか。

２，それぞれの勉強時間（studytime）についてはどうでしょうか。

In [1]:
import pandas as pd
from scipy import stats

# データ読み込み
student_data_math = pd.read_csv("student/student-mat.csv", sep=";")
student_data_por = pd.read_csv("student/student-por.csv", sep=";")

# 同じ生徒を対応させるための結合キー
merge_keys = [
    "school", "sex", "age", "address", "famsize", "Pstatus",
    "Medu", "Fedu", "Mjob", "Fjob", "reason", "nursery", "internet"
]

# 数学データとポルトガル語データを結合
student_data_merge = pd.merge(
    student_data_math,
    student_data_por,
    on=merge_keys,
    suffixes=("_math", "_por")
)

print("結合後の人数:", len(student_data_merge))

# 欠席数 absences の対応のあるt検定
t_abs, p_abs = stats.ttest_rel(
    student_data_merge["absences_math"],
    student_data_merge["absences_por"]
)

print("数学 absences 平均:", student_data_merge["absences_math"].mean())
print("ポルトガル語 absences 平均:", student_data_merge["absences_por"].mean())
print("absences t値:", t_abs)
print("absences p値:", p_abs)

# 勉強時間 studytime の対応のあるt検定
t_study, p_study = stats.ttest_rel(
    student_data_merge["studytime_math"],
    student_data_merge["studytime_por"]
)

print("数学 studytime 平均:", student_data_merge["studytime_math"].mean())
print("ポルトガル語 studytime 平均:", student_data_merge["studytime_por"].mean())
print("studytime t値:", t_study)
print("studytime p値:", p_study)

結合後の人数: 382
数学 absences 平均: 5.319371727748691
ポルトガル語 absences 平均: 3.6727748691099475
absences t値: 4.794359243088911
absences p値: 2.344165688838404e-06
数学 studytime 平均: 2.0340314136125652
ポルトガル語 studytime 平均: 2.0392670157068062
studytime t値: -0.5768458127637375
studytime p値: 0.5643842756976524


有意水準5%で対応のあるt検定を行った。

欠席数 absences について、数学の平均は5.319、ポルトガル語の平均は3.673であった。
検定の結果、p値は2.34×10^-6となり、0.05より小さかった。
したがって、数学とポルトガル語の欠席数には差があると言える。

勉強時間 studytime について、数学の平均は2.034、ポルトガル語の平均は2.039であった。
検定の結果、p値は0.564となり、0.05より大きかった。
したがって、数学とポルトガル語の勉強時間には差があるとは言えない。